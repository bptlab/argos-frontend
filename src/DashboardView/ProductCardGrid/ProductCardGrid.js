import React, { Component } from 'react';
import './ProductCardGrid.css';
import ProductCard from './ProductCard/ProductCard.js';

class ProductCardGrid extends Component {
    searchMatches(product) {
        if (!this.props.searchText) {
            return true;
        }
        //SEARCH-CONFIG: Edit this to define relevant fields for overview search
        const searchFields = [
            product.name,
            product.orderNumber,
            product.productionStart
        ];

        return !searchFields.every((searchField) => {
            return !(searchField.toString().toLowerCase().indexOf(this.props.searchText) > -1);
        });
    }

    render() {
        return (
            <div className="container">
                <div className="row">
                    {this.props.products.map((product) =>  {
                        if(this.searchMatches(product)) {
                            return (
                                <ProductCard
                                    key={product.id}
                                    product={product}/>
                            );
                        } else {
                            return false;
                        }
                    })}
                </div>
            </div>
        );
    }
}
export default ProductCardGrid;