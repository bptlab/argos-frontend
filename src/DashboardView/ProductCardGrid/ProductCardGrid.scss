.ProductCardGrid {
    display: flex;
    align-items: flex-start;
    flex-wrap: wrap;
    justify-content: space-between;
}