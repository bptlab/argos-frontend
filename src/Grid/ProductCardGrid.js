import React, { Component } from 'react';
import './ProductCardGrid.scss';

class Grid extends Component {
    render() {
        return (
            <div className="ProductCardGrid">
                {this.props.children}
            </div>
        );
    }
}

export default Grid;
