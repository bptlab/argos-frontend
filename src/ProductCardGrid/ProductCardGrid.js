import React, { Component } from 'react';
import './ProductCardGrid.css';
import ProductCard from '../ProductCard/ProductCard.js';

class ProductCardGrid extends Component {

    searchMatches(product) {
        if (!this.props.searchText) {
            return true;
        }
        let searchFields = [product.metaData.brand, product.metaData.label, product.metaData.orderNumber,
            product.productionStart];
        for(let index in searchFields) {
            if (searchFields[index].toString().indexOf(this.props.searchText) > -1) {
                return true;
            }
        }
        return false;
    }

    render() {
        return (
            <div className="ProductCardGrid">
                {this.props.products.map((product) =>  {
                    if(this.searchMatches(product)) {
                        return (<ProductCard key={product.id} product={product}/>);
                    } else {
                        return false;
                    }
                })}
            </div>
        );
    }
}

export default ProductCardGrid;
