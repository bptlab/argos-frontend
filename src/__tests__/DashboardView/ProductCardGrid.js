import React from 'react';
import ProductCardGrid from '../../DashboardView/ProductCardGrid/ProductCardGrid.js';
import renderer from 'react-test-renderer';
import ProductData from '../testData/products.js';
import ProductFamilyData from '../testData/frontend_productFamilies';
let instance, component;

beforeEach(() => {
    component = renderer.create(
        <ProductCardGrid
            ref={(child) => {instance = child}}
            productFamilies={ProductFamilyData.PRODUCTFAMILIES}
            excludedStates={[]}/>
    );
});

test("Correct drawing of ProductCardGrid", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Correct drawing of ProductCardGrid in case of excluded state", () => {
    const component2 = renderer.create(
        <ProductCardGrid
            productFamilies={ProductFamilyData.PRODUCTFAMILIES}
            excludedStates={["WARNING"]}/>
    );
    const tree2 = component2.toJSON();
    expect(tree2).toMatchSnapshot();
});

test("Correct search of products", () => {
    let testResult = instance.searchMatches(ProductData.PRODUCTS[0], ProductData.PRODUCTS[0].name);
    expect(testResult).toEqual(true);
    testResult = instance.searchMatches(ProductData.PRODUCTS[0], "Something unreal: #+343");
    expect(testResult).toEqual(false);
});