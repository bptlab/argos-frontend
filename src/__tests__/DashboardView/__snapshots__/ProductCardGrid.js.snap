exports[`test Correct drawing of ProductCardGrid 1`] = `
<div
  className="container">
  <div
    id="product-family-accordion"
    role="tablist">
    <div
      className="card">
      <a
        className=""
        data-parent="#product-family-accordion"
        data-toggle="collapse"
        href="#collapse-product-family-0">
        <div
          className="card-header"
          id="product-family-header-0"
          role="tab">
          <h5
            className="mb-0">
            Testproductfamily
          </h5>
        </div>
      </a>
      <div
        className="collapse show"
        id="collapse-product-family-0"
        role="tabpanel">
        <div
          className="card-block">
          <div
            className="row">
            <div
              className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
              <a
                href="/#/product/0"
                title="Details for Testproduct 5000 14E-17B">
                <div
                  className="card ProductCard ProductCard_RUNNING">
                  <h4
                    className="card-title ProductCardTitle ProductCardTitle_RUNNING">
                    Testproduct 5000 14E-17B
                  </h4>
                  <ul>
                    <li
                      className="ProductCardInformationLine PCIL_serialNumber">
                      9830510416
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_quantity">
                      52189
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_date">
                      2016-05-07
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_errorQuantity">
                      84
                    </li>
                  </ul>
                </div>
              </a>
            </div>
            <div
              className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
              <a
                href="/#/product/1"
                title="Details for Testproduct 5010 116-1EB">
                <div
                  className="card ProductCard ProductCard_WARNING">
                  <h4
                    className="card-title ProductCardTitle ProductCardTitle_WARNING">
                    Testproduct 5010 116-1EB
                  </h4>
                  <ul>
                    <li
                      className="ProductCardInformationLine PCIL_serialNumber">
                      7716013254
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_quantity">
                      90621
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_date">
                      2016-04-06
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_errorQuantity">
                      741
                    </li>
                  </ul>
                </div>
              </a>
            </div>
            <div
              className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
              <a
                href="/#/product/2"
                title="Details for Testproduct 3010 226-1XC">
                <div
                  className="card ProductCard ProductCard_WARNING">
                  <h4
                    className="card-title ProductCardTitle ProductCardTitle_WARNING">
                    Testproduct 3010 226-1XC
                  </h4>
                  <ul>
                    <li
                      className="ProductCardInformationLine PCIL_serialNumber">
                      7710473416
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_quantity">
                      746
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_date">
                      2016-01-01
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_errorQuantity">
                      362
                    </li>
                  </ul>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card">
      <a
        className="collapsed"
        data-parent="#product-family-accordion"
        data-toggle="collapse"
        href="#collapse-product-family-1">
        <div
          className="card-header"
          id="product-family-header-1"
          role="tab">
          <h5
            className="mb-0">
            TestproductfamilyNeo
          </h5>
        </div>
      </a>
      <div
        className="collapse"
        id="collapse-product-family-1"
        role="tabpanel">
        <div
          className="card-block">
          <div
            className="row">
            <div
              className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
              <a
                href="/#/product/5"
                title="Details for Testproduct 7000 27E-11C">
                <div
                  className="card ProductCard ProductCard_WARNING">
                  <h4
                    className="card-title ProductCardTitle ProductCardTitle_WARNING">
                    Testproduct 7000 27E-11C
                  </h4>
                  <ul>
                    <li
                      className="ProductCardInformationLine PCIL_serialNumber">
                      48958959
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_quantity">
                      3475
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_date">
                      2016-02-06
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_errorQuantity">
                      13
                    </li>
                  </ul>
                </div>
              </a>
            </div>
            <div
              className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
              <a
                href="/#/product/6"
                title="Details for Testproduct 5010 115">
                <div
                  className="card ProductCard ProductCard_RUNNING">
                  <h4
                    className="card-title ProductCardTitle ProductCardTitle_RUNNING">
                    Testproduct 5010 115
                  </h4>
                  <ul>
                    <li
                      className="ProductCardInformationLine PCIL_serialNumber">
                      7716013254
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_quantity">
                      90621
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_date">
                      2016-04-06
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_errorQuantity">
                      741
                    </li>
                  </ul>
                </div>
              </a>
            </div>
            <div
              className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
              <a
                href="/#/product/7"
                title="Details for Testproduct 012 226-1FC">
                <div
                  className="card ProductCard ProductCard_ERROR">
                  <h4
                    className="card-title ProductCardTitle ProductCardTitle_ERROR">
                    Testproduct 012 226-1FC
                  </h4>
                  <ul>
                    <li
                      className="ProductCardInformationLine PCIL_serialNumber">
                      7710473416
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_quantity">
                      746
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_date">
                      2016-02-01
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_errorQuantity">
                      362
                    </li>
                  </ul>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test Correct drawing of ProductCardGrid in case of excluded state 1`] = `
<div
  className="container">
  <div
    id="product-family-accordion"
    role="tablist">
    <div
      className="card">
      <a
        className=""
        data-parent="#product-family-accordion"
        data-toggle="collapse"
        href="#collapse-product-family-0">
        <div
          className="card-header"
          id="product-family-header-0"
          role="tab">
          <h5
            className="mb-0">
            Testproductfamily
          </h5>
        </div>
      </a>
      <div
        className="collapse show"
        id="collapse-product-family-0"
        role="tabpanel">
        <div
          className="card-block">
          <div
            className="row">
            <div
              className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
              <a
                href="/#/product/0"
                title="Details for Testproduct 5000 14E-17B">
                <div
                  className="card ProductCard ProductCard_RUNNING">
                  <h4
                    className="card-title ProductCardTitle ProductCardTitle_RUNNING">
                    Testproduct 5000 14E-17B
                  </h4>
                  <ul>
                    <li
                      className="ProductCardInformationLine PCIL_serialNumber">
                      9830510416
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_quantity">
                      52189
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_date">
                      2016-05-07
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_errorQuantity">
                      84
                    </li>
                  </ul>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card">
      <a
        className="collapsed"
        data-parent="#product-family-accordion"
        data-toggle="collapse"
        href="#collapse-product-family-1">
        <div
          className="card-header"
          id="product-family-header-1"
          role="tab">
          <h5
            className="mb-0">
            TestproductfamilyNeo
          </h5>
        </div>
      </a>
      <div
        className="collapse"
        id="collapse-product-family-1"
        role="tabpanel">
        <div
          className="card-block">
          <div
            className="row">
            <div
              className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
              <a
                href="/#/product/6"
                title="Details for Testproduct 5010 115">
                <div
                  className="card ProductCard ProductCard_RUNNING">
                  <h4
                    className="card-title ProductCardTitle ProductCardTitle_RUNNING">
                    Testproduct 5010 115
                  </h4>
                  <ul>
                    <li
                      className="ProductCardInformationLine PCIL_serialNumber">
                      7716013254
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_quantity">
                      90621
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_date">
                      2016-04-06
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_errorQuantity">
                      741
                    </li>
                  </ul>
                </div>
              </a>
            </div>
            <div
              className="col-xs-12 col-sm-6 col-md-6 col-lg-3 col-uhd-2">
              <a
                href="/#/product/7"
                title="Details for Testproduct 012 226-1FC">
                <div
                  className="card ProductCard ProductCard_ERROR">
                  <h4
                    className="card-title ProductCardTitle ProductCardTitle_ERROR">
                    Testproduct 012 226-1FC
                  </h4>
                  <ul>
                    <li
                      className="ProductCardInformationLine PCIL_serialNumber">
                      7710473416
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_quantity">
                      746
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_date">
                      2016-02-01
                    </li>
                    <li
                      className="ProductCardInformationLine PCIL_errorQuantity">
                      362
                    </li>
                  </ul>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
